scripts/validation/validate-versions-staged-files.js

```diff
@@ -0,0 +1,62 @@
+const { join, dirname } = require("path");
+const { promisify } = require("util");
+const { exec, execFile } = require("child_process");
+
+const execAsync = promisify(exec);
+const execFileAsync = promisify(execFile);
+
+preCommit().catch(error => {
+    console.error(error);
+    process.exit(1);
+});
+
+async function preCommit() {
+    const [{ stdout: lernaPackages }, { stdout: stagedFiles }] = await Promise.all([
+        execAsync("./node_modules/.bin/lerna ls --json --all"),
+        execAsync("git diff --staged --name-only")
+    ]);
+    const packages = JSON.parse(lernaPackages.trim());
+    const staged = stagedFiles.trim().split("\n");
+    const changedWidgetPackages = packages
+        .filter(({ location }) =>
+            staged.some(
+                changedFilePath =>
+                    dirname(join(process.cwd(), changedFilePath)).includes(location) &&
+                    changedFilePath
+                        .split("/")
+                        .pop()
+                        .match(/package\.(json|xml)$/)
+            )
+        )
+        .filter(({ location }) => location.match(/(pluggable|custom)Widgets/));
+
+    if (changedWidgetPackages.length) {
+        const validationPromises = [];
+
+        for (const changedWidgetPackage of changedWidgetPackages) {
+            validationPromises.push(
+                new Promise((resolve, reject) => {
+                    execFileAsync("node", [join(process.cwd(), "scripts/validation/validate-version.js")], {
```

Comment:
Is there a reason why we moved "validate-version.js" to a different file ? If this the only place lets inline it !

Reply:
Makes sense to inline 👍  Changed